virtualenv/builders/legacy.py

```diff
@@ -0,0 +1,228 @@
+# paths that the interpreter adds itself. These are essentially the locations
+# it looks for the various stdlib modules. Since we are inside of a virtual
+# environment these will all be relative to the sys.prefix and sys.exec_prefix,
+# however we want to change these to be relative to sys.base_prefix and
+# sys.base_exec_prefix instead.
+new_sys_path = []
+for path in sys.path:
+    # TODO: Is there a better way to determine this?
+    if path.startswith(sys.prefix):
+        path = os.path.join(
+            sys.base_prefix,
+            path[len(sys.prefix):],
+        )
+    elif path.startswith(sys.exec_prefix):
+        path = os.path.join(
+            sys.base_exec_prefix,
+            path[len(sys.exec_prefix):],
+        )
+
+    new_sys_path.append(path)
+sys.path = new_sys_path
+
+# We want to empty everything that has already been imported from the
+# sys.modules so that any additional imports of these modules will import them
+# from the base Python and not from the copies inside of the virtual
+# environment. This will ensure that our copies will only be used for
+# bootstrapping the virtual environment.
+for key in list(sys.modules):
+    # We don't want to purge these modules because if we do, then things break
+    # very badly.
+    if key in ["sys", "site", "sitecustomize", "__builtin__", "__main__"]:
+        continue
+
+    del sys.modules[key]
+
+# We want to trick the interpreter into thinking that the user specific
+# site-packages has been requested to be disabled. We'll do this by mimicing
+# that sys.flags.no_user_site has been set to False, however sys.flags is a
+# read-only structure so we'll temporarily replace it with one that has the
+# same values except for sys.flags.no_user_site which will be set to True.
+_real_sys_flags = sys.flags
+class AttrDict(dict):
+    def __init__(self, *args, **kwargs):
+        dict.__init__(self, *args, **kwargs)
+    def __getattr__(self, name):
+        return self[name]
+sys.flags = AttrDict((k, getattr(sys.flags, k)) for k in dir(sys.flags))
+sys.flags["no_user_site"] = True
+
+# We want to import the *real* site module from the base Python. Actually
+# attempting to do an import here will just import this module again, so we'll
+# just read the real site module and exec it.
+with open(__SITE__) as fp:
+    exec(fp.read())
+
+# Restore the real sys.flags
+sys.flags = _real_sys_flags
+
+# If we're running with the global site-packages enabled, then we'll want to
+# go ahead and enable it here so that it comes after the virtual environment's
+# site-package.
+if global_site_packages is not None:
+    # Force a re-check of ENABLE_USER_SITE so that we get the "real" value
+    # instead of our forced off value.
+    ENABLE_USER_SITE = check_enableusersite()
+
+    # Add the actual user site-packages if we're supposed to.
+    addusersitepackages(None)
+
+    # Add the actual global site-packages.
+    for path in global_site_packages:
+        addsitedir(path)
```

Comment:
I think you may still need the function `force_global_eggs_after_local_site_packages` from the old `site.py`. Otherwise eggs will push their paths to the top of `sys.path`, causing globally-installed eggs to override venv-installed non-eggs for the same package. This would be a regression.